compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1805,15 +1805,17 @@ fn receiver_is_valid<'tcx>(
 
     confirm_type_is_not_a_method_generic_param(receiver_ty, method_generics)?;
 
-    let mut autoderef = Autoderef::new(infcx, wfcx.param_env, wfcx.body_def_id, span, receiver_ty);
+    let hard_coded_receiver_trait_def_id =
+        tcx.require_lang_item(LangItem::LegacyReceiver, Some(span));
```

Comment:
It was correct, in that this is only used in the code path without the `arbitrary_self_types` feature enabled, but I've moved it so that we don't even look it up in the arbitrary-self-types case.